core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -849,4 +850,31 @@ private String toStringSlow(@Nullable Channel ch, @Nullable RequestLogAccess par
             return buf.toString();
         }
     }
+
+    @Override
+    public CompletableFuture<Void> initiateConnectionShutdown() {
+        final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+
+        final Channel ch = channel();
+        if (ch == null || !ch.isActive()) {
+            setAdditionalRequestHeader(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+            log().whenComplete().whenComplete((ignore, ex) -> {
+                if (ex == null) {
+                    completableFuture.complete(null);
+                } else {
+                    completableFuture.completeExceptionally(ex);
+                }
```

Comment:
Can we check `channel()` is not `null` here again and use `ch.closeFuture()` if possible? If the `channel()` is not available even at this point, we could simply complete the future as you did here.

Reply:
Thanks for your advice